### PR TITLE
Make sure we sync any identity contracts we haven't synced upon login

### LINKED ISSUE #1684: Make sure we sync any identity contracts we haven't sync'd upon login
### Problem

In the sideEffect for `'gi.contracts/group/inviteAccept'`, it's possible that we fail to sync user contracts for some reason or another (a server could be down, etc.):

https://github.com/okTurtles/group-income/commit/0c1634b7343dc5d27fecc241506fcb70dfc50917#r123335604

### Solution

Upon logging in, check the users we have in our group, and use `'namespace/lookup'` to see if there are any identity contracts that we haven't sync'd, and sync those.

### PR DESCRIPTION
**Pull request by Silver-IT**
### Summary of Changes

- Closes #1684

### CODE REVIEW

File: frontend/controller/actions/identity.js [hunk 1]
```diff
@@ -318,10 +318,7 @@ export default (sbp('sbp/selectors/register', {
       const additionalIdentityContractIDs = await Promise.all(chatRoomUsers.filter(username => {
         return getters.ourUsername !== username && !getters.ourContacts.includes(username)
       }).map(username => sbp('namespace/lookup', username)))
-
-      for (const identityContractID of additionalIdentityContractIDs) {
-        await sbp('chelonia/contract/sync', identityContractID)
-      }
```
Comment (taoeffect) [inline frontend/controller/actions/identity.js:324]: This is a nice simplification 👍

File: frontend/controller/actions/group.js [hunk 2]
```diff
@@ -316,7 +316,9 @@ export default (sbp('sbp/selectors/register', {
   // action if we haven't done so yet (because we were previously waiting for
   // the keys), or (c) already a member and ready to interact with the group.
   'gi.actions/group/join': async function (params: $Exact<ChelKeyRequestParams> & { options?: { skipUsableKeysCheck?: boolean; skipInviteAccept?: boolean } }) {
-    sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, true)
+    if (!params.options?.skipInviteAccept) {
```
Comment (taoeffect) [inline frontend/controller/actions/group.js]: Why was this change added?

And I feel like this `skipInviteAccept` must have been removed for some reason, maybe @corrideat can also comment on that?

File: frontend/controller/actions/group.js [hunk 3]
```diff
@@ -512,6 +514,13 @@ export default (sbp('sbp/selectors/register', {
           })
         }
 
+        // NOTE: sync identity contracts which are out of sync after joining group
+        const missingIDs = (await Promise.all(
+          Object.keys(state.profiles)
+            .map(username => sbp('namespace/lookup', username))
+        )).filter(id => !rootState[id] && !sbp('chelonia/contract/isSyncing', id))
+        await sbp('chelonia/contract/sync', missingIDs)
```
Comment (taoeffect) [inline frontend/controller/actions/group.js]: I recommend wrapping this in an `if` to check if `missingIDs.length > 0`, and then you can also add a line to: `console.info('found unsynced identity contracts to sync:', missingIDs)`

**Review by taoeffect — CHANGES_REQUESTED** [review]
Nice work @Silver-IT!

Minor change requests + questions for you and @corrideat

File: frontend/controller/actions/group.js (hunk 3 above)
Comment (Silver-IT) [reply frontend/controller/actions/group.js]: Got it.

File: frontend/controller/actions/group.js (hunk 2 above)
Comment (Silver-IT) [reply frontend/controller/actions/group.js]: Actually, `JOINING_GROUP-params.contractID` was designed to be used in `sideEffect` functions only when joining group, not when logging in. It provide us the way to ignore removing group contract if we meet `removeOurselves` while joining group. `skipInviteAccept` is true only when login.

Having investigated the `gi.actions/group/join` function and login process, I have noticed that we sync the group contracts many times while logging in. And it's not easy to decide when should set this `JOINING_GROUP-params.contractID` true. If we remove the `if` conditional statement, there won't be any errors, but the contract would be removed and synced which is redundant activity.
At the moment, I am not sure when should I set, which conditional check would be correct, so I will remove this `if` conditional statement.

However, this `JOINING_GROUP-params.contractID` should be set `true` only when we sync the group contract to join.

**Review by taoeffect — APPROVED** [review]
Nicely done 👍

**Pull request merged**